modules/core/src/main/java/org/apache/synapse/util/xpath/DecryptFunction.java

```diff
@@ -0,0 +1,123 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.synapse.util.xpath;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jaxen.Context;
+import org.jaxen.Function;
+import org.jaxen.FunctionCallException;
+import org.jaxen.function.StringFunction;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.util.Base64;
+import java.util.List;
+
+/*
+Xpath function to decrypt based on keystore on Asymmetric keys
+ */
+public class DecryptFunction implements Function {
+    private static final Log log = LogFactory.getLog(DecryptFunction.class);
+    private static final String DEFAULT_ALGORITHM = "RSA";
+    private static final String DEFAULT_KEYSTORE_TYPE = "JKS";
+
+    @Override
+    public Object call(Context context, List args) throws FunctionCallException {
+        boolean debugOn = log.isDebugEnabled();
+
+        if (args == null) {
+            if (debugOn) {
+                log.debug("Missing arguments in the function call");
+            }
+            return SynapseXPathConstants.NULL_STRING;
+        }
```

Comment:
we can skip this if condition as this will anyway return null in the end

Reply:
This is to avoid NPE when we do args.size()